### ISSUE TITLE
Schematic based build system

### ISSUE THREAD

**Issue by mattweil** [michaelzangl/minebot#79]
Minebot's build system is quite complicated, it works from some sort of queue system if I am not mistaken... I am unsure of how difficult this would be to implement but a schematic based build system in which you simply type /minebot schem [name].schematic would be extremely convenient and simple to use

**Comment by CarbonNeuron** [comment]
Minebot is a schedule based block placer the only thing Michael would need to do is have the schematic get sliced and get each block and then run that through the scheduling system

**Comment by michaelzangl** [comment]
Yes, that would be most of it and I already looked into integrating this with [Schematica](https://github.com/Lunatrius/Schematica). It would also allow for a better preview.

Currently, I do not have the time to do it.

**Comment by mossyblog** [comment]
Well, I've been tinkering around and from what I've got so far is a possible connection point here.

Using Schematica Printing mode you can combine the two without really having to know about each other.

1. Position Schematic in a spot you deem suitable.
2. Turn on Printer mode and then choose "layer" instead of "show all"
3. MineBot Pos1 and Pos2 match the two corners of your "boxed" schematic.
4. Run script that both clears that boxed area and moves in a grid fashion 1x YPOS at time.

This will get you essentially 90% of the automation done. For blocks like logs facing etc there are some issues in that you may have to jumpstart the "printing" manually (ie Schematica still relies on adjacent placement(s)).  The other issue is things like stairs facing in opposite direction of travel but this could be fixed by making Minebot reverse direction of travel?

I think one could probably keep the two source codes separate. The only issue I have at the moment is making Minebot move in a grid fashion on two Pos1 and Pos2 points as refrerence.
